### PR TITLE
[Serializer] Remove unused code

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php [hunk 1]
```diff
@@ -176,7 +140,7 @@ protected function setAttributeValue($object, $attribute, $value, $format = null
     {
         $setter = 'set'.ucfirst($attribute);
 
-        if (is_callable(array($object, $setter))) {
+        if (is_callable(array($object, $setter)) && !(new \ReflectionMethod($object, $setter))->isStatic()) {
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php]: What about caching the information gathered through reflection?

**Comment by dunglas** [comment]
@xabbuh cache added, good idea thanks!

File: src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php [hunk 2]
```diff
@@ -175,8 +141,13 @@ protected function getAttributeValue($object, $attribute, $format = null, array
     protected function setAttributeValue($object, $attribute, $value, $format = null, array $context = array())
     {
         $setter = 'set'.ucfirst($attribute);
+        $key = get_class($object).':'.$setter;
+
+        if (!isset(self::$setterAccessibleCache[$setter])) {
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php]: `$setter` should be `$key`

Comment (dunglas) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php]: Fixed

**Comment by Tobion** [comment]
:+1: 
Status: Reviewed

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @dunglas.

**Pull request closed without merging**